model_analyzer/monitor/nvml.py

```diff
@@ -66,23 +66,26 @@ def _monitoring_loop(self, tags):
         self._thread_active = True
         frequency = self._frequency
 
-        record_collector = RecordCollector()
+        records = []
         while self._thread_active:
             begin = time.time()
             for tag in tags:
                 if tag == 'memory':
                     for i, handle in enumerate(self._nvml_handles):
                         memory = nvmlDeviceGetMemoryInfo(handle)
                         gpu_device = self._gpus[i]
-                        memory_record = GPUMemoryRecord(
-                            gpu_device, memory.used, memory.total, memory.free)
-                        record_collector.insert(memory_record)
+                        records.append(
+                            GPUUsedMemory(device=gpu_device,
+                                          used_mem=memory.used))
+                        records.append(
+                            GPUFreeMemory(device=gpu_device,
+                                          free_mem=memory.free))
```

Comment:
`GPUTotalMemory` seems to be missing here.

Reply:
I was wondering whether we need that or whether we could just compute it and pass it if we needed it. Is it not just GPUFreeMemory + GPUUsedMemory? Seemed like a redundant record.

Reply:
Right, it may be redundant.